lib/node_modules/@stdlib/repl/lib/syntax_highlighter.js

```diff
@@ -125,6 +140,138 @@ setNonEnumerableReadOnly( SyntaxHighlighter.prototype, '_highlightLine', functio
 	return highlightedLine;
 });
 
+/**
+* Returns a list of available theme names.
+*
+* @name getThemes
+* @memberof SyntaxHighlighter.prototype
+* @type {Function}
+* @returns {Array} array of all theme names
+*/
+setNonEnumerableReadOnly( SyntaxHighlighter.prototype, 'getThemes', function getThemes() {
+	return objectKeys( this._themes );
+});
+
+/**
+* Sets the current color theme.
+*
+* @name setTheme
+* @memberof SyntaxHighlighter.prototype
+* @type {Function}
+* @param {string} theme - theme name
+* @throws {Error} argument must be an existing theme
+*/
+setNonEnumerableReadOnly( SyntaxHighlighter.prototype, 'setTheme', function setTheme( theme ) {
+	if ( !contains( objectKeys( this._themes ), theme ) ) {
+		throw new Error( format( 'invalid argument. First argument must be an existing theme name. Value: `%s`.', theme ) );
+	}
+	this._highlightedLine = '';
+	this._theme = theme;
+});
+
+/**
+* Returns a theme's colors palette.
+*
+* @name getThemeConfig
+* @memberof SyntaxHighlighter.prototype
+* @type {Function}
+* @param {string} theme - theme name
+* @throws {Error} must provide an existing theme name
+* @returns {Object} theme object
+*/
+setNonEnumerableReadOnly( SyntaxHighlighter.prototype, 'getThemeConfig', function getThemeConfig( theme ) {
+	if ( !contains( objectKeys( this._themes ), theme ) ) {
+		throw new Error( format( 'invalid argument. First argument must be an existing theme name. Value: `%s`.', theme ) );
+	}
+	return this._themes[ theme ];
+});
+
+/**
+* Adds a new color theme.
+*
+* @name addTheme
+* @memberof SyntaxHighlighter.prototype
+* @type {Function}
+* @param {Object} theme - theme object
+*/
+setNonEnumerableReadOnly( SyntaxHighlighter.prototype, 'addTheme', function addTheme( name, theme ) {
+	this._themes[ name ] = theme;
```

Comment:
Based on this logic, we are okay with overwriting/clobbering existing themes. That's probably fine; it's a shortcut for `deleteTheme` + `addTheme`. But I wanted to highlight that this is intentional choice.